Store Orders: Add activity log (order notes) back to single orders

ryelle opened the pull request:
Fixes #13687 

This PR brings the Activity log card back to the single order view, and in it lists out the notes associated with an order. These are status updates on the order, saved as comments on the order. They show updates to the order status, such as when a transaction succeeds/fails, stock is reduced, a refunded granted, etc.

There is no indication in the content of the note to say what kind of note it is, except whether it was sent to the customer or not. The design has different icons for refunding & labels, but I don't think we'll be able to do that. I was going to try a few ways of testing the note content to check the "type", but notes are also translated into the admin's language, so if I search for "Refund ID", it'll only work for English sites.

Pretty sure this'll also mean we can't add the refund label/reprint label actions. But I can add "resend email" to any notes that were emailed to the customer.

![screen shot 2017-08-15 at 3 59 40 pm](https://user-images.githubusercontent.com/541093/29335607-42f2b6f2-81d9-11e7-992d-1381697b0d22.png)

These two states could use a design 👀  :

Loading
![screen shot 2017-08-15 at 4 36 20 pm](https://user-images.githubusercontent.com/541093/29335605-42e976d2-81d9-11e7-8b8e-93a89475c066.png)

No notes (this would not show by default, the user would have to delete the automated "order created" notes)
![screen shot 2017-08-15 at 4 35 35 pm](https://user-images.githubusercontent.com/541093/29335606-42ebc9be-81d9-11e7-8889-cac8aa27a928.png)

**To test**

- View an order
- There should be two boxes under the order details now (Activity Log & Customer Info)
- The Activity Log should load in pretty quickly, and you'll see any notes on the order (probably at least something like "Order status changed from Pending payment to Processing.")
- Check this against the notes list in wp-admin to make sure nothing's missing.
- You can create a customer note by fulfilling the order using the Fulfill modal, and emailing the user a tracking number.

a8ci18n commented on client/extensions/woocommerce/app/order/order-notes/index.js:
```diff
@@ -1,34 +1,127 @@
+		const { days, notesByDay, translate } = this.props;
+		if ( ! days.length ) {
+			return (
+				<p>{ translate( 'No activity yet' ) }</p>
```
Hi! I've found a possible matching string that has already been translated **83** times:
`translate( 'No activity yet!' )` *ES Score: 15*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/store-orders-view-notes)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

kellychoffman commented:
Looks awesome!

> Loading

Could we use loading indicators? Like: 

<img width="438" alt="screen shot 2017-08-16 at 10 04 12 am" src="https://user-images.githubusercontent.com/5835847/29373210-bd66c126-826a-11e7-8388-e1188f306f17.png">

kellychoffman commented:
> No notes (this would not show by default, the user would have to delete the automated "order created" notes)

At first I was thinking we could show the emptyContent here, but I think it might be overkill since we don't have the ability to delete notes in WP.com. @jameskoster: I'm thinking we should just keep the plain text notice, what do you think?

allendav commented on client/extensions/woocommerce/app/order/order-notes/day.js:
```diff
@@ -0,0 +1,52 @@
+
+class OrderNotesByDay extends Component {
+	static propTypes = {
+		count: PropTypes.number,
```
All of these props seem to be required. Should have include isRequired?

allendav commented on client/extensions/woocommerce/app/order/order-notes/index.js:
```diff
@@ -1,34 +1,127 @@
+					count={ notes.length }
+					onClick={ this.toggleOpenDay }
+					isOpen={ index === this.state.openIndex }>
+					{ notes.map( note => <OrderNote { ...note } key={ note.id } /> ) }
```
Nice :)

allendav commented on client/extensions/woocommerce/app/order/order-notes/note.js:
```diff
@@ -0,0 +1,57 @@
+		return (
+			<div className="order-notes__note">
+				<div className="order-notes__note-meta">
+					<span className="order-notes__note-time">{ moment( date_created ).format( 'LT' ) }</span>
```
Should probably use date_created_gmt instead, to avoid browser timezone != store timezone issues.  You'll need to append Z of course before passing to moment.  See here for an example: https://github.com/Automattic/wp-calypso/blob/master/client/extensions/woocommerce/app/order/order-created/index.js#L22

allendav commented on client/extensions/woocommerce/app/order/order-notes/note.js:
```diff
@@ -0,0 +1,57 @@
+	static propTypes = {
+		customer_note: PropTypes.bool,
+		date_created: PropTypes.string,
+		date_created_gmt: PropTypes.string,
```
note and date_created_gmt (see comment below) should probably be isRequired

allendav commented on client/extensions/woocommerce/app/order/order-notes/note.js:
```diff
@@ -0,0 +1,57 @@
+		customer_note: PropTypes.bool,
+		date_created: PropTypes.string,
+		date_created_gmt: PropTypes.string,
+		id: PropTypes.number,
```
I don't think you use id in this component

allendav commented on client/extensions/woocommerce/app/order/order-notes/style.scss:
```diff
@@ -0,0 +1,63 @@
+
+	&:before {
+		content: '';
+		z-index: -1;
```
Do we usually alphabetize CSS or not really?

allendav commented:
Should probably be "event" and not "events" when there is just one

<img width="503" alt="screen shot 2017-08-16 at 11 45 20 am" src="https://user-images.githubusercontent.com/1595739/29379863-a568f982-8278-11e7-9a7a-6981aa3952aa.png">

allendav commented:
Could we use the `ll` format for date instead - in order to be consistent with dates in the order list and in individual orders, i.e.:

<img width="663" alt="screen shot 2017-08-16 at 11 50 13 am" src="https://user-images.githubusercontent.com/1595739/29380054-4f273452-8279-11e7-901a-3803b27421ef.png">

<img width="662" alt="screen shot 2017-08-16 at 11 50 04 am" src="https://user-images.githubusercontent.com/1595739/29380053-4f239d88-8279-11e7-9e18-f41d46482526.png">

ryelle commented:
I've added a loading indicator (and addressed @allendav's feedback). The only problem (maybe?) with the loading indicator is that it fades in opacity, so at the lowest "pulse" you can see the line through the content:
<img width="441" alt="screen shot 2017-08-16 at 3 38 13 pm" src="https://user-images.githubusercontent.com/541093/29382186-d3b44988-8299-11e7-9a83-c6ab2c60c3d4.png">

Maybe not an issue, since theoretically they shouldn't see this for long enough to really register it, but I noticed in testing.

kellychoffman commented:
> Maybe not an issue, since theoretically they shouldn't see this for long enough to really register it, but I noticed in testing.

I think it's okay.

jameskoster commented:
> @jameskoster: I'm thinking we should just keep the plain text notice, what do you think?

@kellychoffman I don't think this would ever display? It's not possible to delete system-generated notes, right?

kellychoffman commented:
> I don't think this would ever display? It's not possible to delete system-generated notes, right?

Not in WP.com, no. I thought you could in wp-admin, which is why I was thinking this plain next notice would be fine.

jameskoster commented:
> I thought you could in wp-admin

TIL. 

Yeah, plain text notice is fine imo.

jameskoster commented:
> You can create a customer note by fulfilling the order using the Fulfill modal, and emailing the user a tracking number.

Seems that I had to refresh for the new note to appear. I think that's a little confusing because the success message appears without a refresh. Can we update the log inline as well?

ryelle commented:
> Can we update the log inline as well?

Sure, I'll get that in a following PR today, but I'm going to merge this one now.
